_posts/2016-08-18-emerging-docker.md

```diff
@@ -0,0 +1,16 @@
+
+Docker provides so-called software images that package software. These images can then be executed inside a docker container. Furthermore, Docker provides a service called [DockerHub](https://hub.docker.com/) that allows users to upload and publish their own images. This allows developers to bundle their software and distribute it easily. Furthermore, since each container runs isolated from the rest of the systems, the developers do not have to rely on any kind of other system configuration.
+
+Within the ODK project this allows us to bundle mathematical software systems (such as [GAP](http://www.gap-system.org/), [Sage](http://www.sagemath.org/), ...) and distribute them in a manner that is both accessible to users and easily maintainable for developers. Furthermore the isolated aspect of docker containers allows us to easily integrate multiple systems together without having to make additional assumptions about the users specific setup -- we can just run all systems in one docker container. This way users can install the entire mathematical software stack that the OpenDreamKit project aims to provide easily.
+
+Docker was originally a linux only application -- it relied on a lot of functionality provided by the Linux kernel. To make it available on Windows and Mac the developers provided a virtual machine that ran a streamlined Linux system with Docker pre-installed. This made it possible for Windows and Mac users to run docker containers,  however it introduced an additional layer of abstraction that came with some disadvantages. The additional virtualisation slowed down docker containers and provided technical limitations when wanting to integrate with the host system. It also required users to install virtualisation software before being able to run any kind of Docker Image -- something that was not very user friendly.
+
+Recently Docker started to build native versions for Windows and Mac. These versions do not rely on linux functionality -- instead they leverage functionality provided by Windows and Mac operating systems natively. In terms of the OpenDreamKit project these are a big step in terms of usability -- they make it significantly easier for users to run a Docker container. Users can now install Docker just as they would install any other software on their machine. Among speed and resource advantages, these versions should make it easier for developers to create docker-powered applications and Docker containers because of better integration between host system and containers,  for example the file system of the physical machine can be mounted inside containers more easily. As a side note, the windows version of Docker still requires some manual configuration -- a setting in the BIOS has to be changed. This is much less effort than was required previously, however it would be much easier if users did not have to do any manual configuration.
```

Comment:
I would say "...  for Windows and OS X" since `OS X` is the official name of the operating system